library/src/main/java/com/google/android/exoplayer/metadata/Id3Parser.java

```diff
@@ -98,6 +136,25 @@ private static int indexOfNot(byte[] data, int fromIndex, byte key) {
     return data.length;
   }
 
+  private static int indexOfEOS(byte[] data, int fromIndex, int encodingByte) {
+    int terminationPos = indexOf(data, fromIndex, (byte) 0);
+
+    // For single byte encoding charsets, we are done
+    if(encodingByte == ID3_TEXT_ENCODING_ISO_8859_1 || encodingByte == ID3_TEXT_ENCODING_UTF_8) {
+      return terminationPos;
+    }
+
+    // Otherwise, look for a two zero bytes
+    while(terminationPos < data.length - 1) {
+      if(data[terminationPos + 1] == (byte) 0) {
+        return terminationPos + 1;
```

Comment:
In fact, I did that intentionally but you are right, I included the termination delimiter in the resulting string but that's wrong and will cause wrong results when converting to string. Thanks for the tip.